trustmanager/keyfilestore.go

```diff
@@ -106,54 +110,97 @@ func (s *KeyMemoryStore) ListKeys() []string {
 	return listKeys(s)
 }
 
-func addKey(s LimitedFileStore, name string, privKey data.PrivateKey) error {
+// RemoveKey removes the key from the keystore
+func (s *KeyMemoryStore) RemoveKey(name string) error {
+	return removeKey(s, name)
+}
+
+
+func addKey(s LimitedFileStore, passphraseRetriever PassphraseRetriever, name, alias string, privKey data.PrivateKey) error {
 	pemPrivKey, err := KeyToPEM(privKey)
 	if err != nil {
 		return err
 	}
 
-	return s.Add(name, pemPrivKey)
-}
+	attempts := 0
+	passphrase := ""
+	giveup := false
+	for {
+		passphrase, giveup, err = passphraseRetriever(name, alias, true, attempts)
+		if err != nil {
+			attempts++
+			continue
+		} else if giveup {
+			return err
+		} else {
+			break
+		}
+	}
 
-func getKey(s LimitedFileStore, name string) (data.PrivateKey, error) {
-	keyBytes, err := s.Get(name)
-	if err != nil {
-		return nil, err
+	if passphrase != "" {
+		pemPrivKey, err = EncryptPrivateKey(privKey, passphrase)
+		if err != nil {
+			return err
+		}
 	}
 
-	// Convert PEM encoded bytes back to a PrivateKey
-	privKey, err := ParsePEMPrivateKey(keyBytes, "")
+	err = s.Add(name + "." + aliasExtension, []byte(alias))
```

Comment:
I thought about suggesting that, but wouldn't that complicate retrieving a key? It would mean you can't just open a file with the key ID as the name; you'd have to list the whole directory and figure out which one matches the key ID.